blocks/v2-all-trucks/v2-all-trucks.js

```diff
@@ -0,0 +1,183 @@
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="none">
+        <path d="M6 4L10 8L6 12" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
+      </svg>
+    </span>
+    `);
+
+  function addShowAllButton() {
+    const showAllButtonLi = createElement('li');
+    const showAllButton = createElement('button',
+      { classes: [`${blockName}__segment-button`, `${blockName}__show-all-button`] });
+
+    showAllButton.textContent = 'All trucks';
+    showAllButtonLi.appendChild(showAllButton);
+    segmentNamesList.appendChild(showAllButtonLi);
+    showAllButton.appendChild(showAllButtonDropDownIcon);
+
+    showAllButton.addEventListener('click', () => {
+      trucks.forEach((truck) => {
+        truck.style.display = 'flex';
+        truck.classList.remove('odd-row', 'even-row');
+      });
+
+      if (MQMobile.matches) {
+        const segmentNamesListWithoutShowAll = document.querySelectorAll(`.${blockName}__button-list > li > button:not(.${blockName}__show-all-button)`);
+        segmentNamesListWithoutShowAll.forEach((item) => {
+          item.classList.toggle('show');
+        });
+      }
+    });
+  }
+
+  addShowAllButton();
+
+  buttonSegmentNames.forEach((segment) => {
+    const li = createElement('li');
+    const filterButton = createElement('button', { classes: `${blockName}__segment-button` });
+    filterButton.textContent = segment;
+    segmentNamesList.appendChild(li);
+    li.append(filterButton);
+
+    filterButton.addEventListener('click', () => {
+      const clickedSegment = filterButton.textContent.trim()
+        .toLowerCase();
+
+      trucks.forEach((truck) => {
+        truck.style.display = truck.classList.contains(clickedSegment) ? 'flex' : 'none';
+        const isSelected = truck.style.display === 'flex';
+        truck.style.display = isSelected ? 'flex' : 'none';
+        truck.classList.toggle('selected-truck', isSelected);
+      });
+
+      const allTrucksRows = document.querySelectorAll(`.${blockName}__truck`);
+
+      allTrucksRows.forEach((truck) => {
+        if (truck.classList.contains('selected-truck')) {
+          truck.classList.remove('odd', 'even');
+
+          const selectedTrucks = document.querySelectorAll('.selected-truck');
+
+          selectedTrucks.forEach((selectedTruck, i) => {
+            if (i % 2 === 0) {
+              selectedTruck.classList.remove('even-row');
+              selectedTruck.classList.add('odd-row');
+            } else {
+              selectedTruck.classList.remove('odd-row');
+              selectedTruck.classList.add('even-row');
+            }
+          });
+        } else if (!truck.classList.contains('selected-truck')) {
+          truck.classList.remove('odd', 'even');
+        }
+      });
+    });
+
+    filterButton.addEventListener('click', () => {
+      const segmentNamesListWithoutShowAll = document.querySelectorAll(`.${blockName}__button-list > li > button:not(.${blockName}__show-all-button)`);
```

Comment:
This code is repeated again for next block at 148, Can make this a function and reuse